sdk/resourcemanager/Azure.ResourceManager.Core/src/SubscriptionOperations.cs

```diff
@@ -100,22 +97,154 @@ public virtual LocationContainer GetLocations()
         /// <inheritdoc/>
         public override ArmResponse<Subscription> Get(CancellationToken cancellationToken = default)
         {
-            return new PhArmResponse<Subscription, ResourceManager.Resources.Models.Subscription>(
-                SubscriptionsClient.Get(Id.Name, cancellationToken),
+            return new PhArmResponse<Subscription, SubscriptionData>(
+                SubscriptionsRestOperations.Get(Id.Name, cancellationToken),
                 Converter());
         }
 
         /// <inheritdoc/>
         public override async Task<ArmResponse<Subscription>> GetAsync(CancellationToken cancellationToken = default)
         {
-            return new PhArmResponse<Subscription, ResourceManager.Resources.Models.Subscription>(
-                await SubscriptionsClient.GetAsync(Id.Name, cancellationToken).ConfigureAwait(false),
+            return new PhArmResponse<Subscription, SubscriptionData>(
+                await SubscriptionsRestOperations.GetAsync(Id.Name, cancellationToken).ConfigureAwait(false),
                 Converter());
         }
 
-        private Func<ResourceManager.Resources.Models.Subscription, Subscription> Converter()
+        /// <summary> Gets all subscriptions for a tenant. </summary>
+        /// <param name="cancellationToken"> The cancellation token to use. </param>
+        public virtual AsyncPageable<Subscription> ListAsync(CancellationToken cancellationToken = default)
```

Comment:
the list methods don't belong here, if you change the private type in container to subscriptionrestOperations this need goes away